test/src/common/MemoryOutputStreamTest.cpp

```diff
@@ -0,0 +1,224 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "string.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "MemoryInputStream.h"
+#include "MemoryOutputStream.h"
+#include "dataBlock.h"
+
+using std::string;
+
+using ::testing::InitGoogleMock;
+using ::testing::InitGoogleTest;
+using testing::Return;
+
+using rocketmq::MemoryBlock;
+using rocketmq::MemoryInputStream;
+using rocketmq::MemoryOutputStream;
+
+TEST(memoryOutputStream, init) {
+    MemoryOutputStream memoryOutput;
+    EXPECT_EQ(memoryOutput.getMemoryBlock().getSize(), 0);
+    EXPECT_TRUE(memoryOutput.getData() != nullptr);
+
+    EXPECT_EQ(memoryOutput.getPosition(), 0);
+    EXPECT_EQ(memoryOutput.getDataSize(), 0);
+
+    MemoryOutputStream twoMemoryOutput(512);
+    EXPECT_EQ(memoryOutput.getMemoryBlock().getSize(), 0);
+
+    MemoryBlock memoryBlock(12, false);
+    MemoryOutputStream threeMemoryOutput(memoryBlock, false);
+    EXPECT_EQ(threeMemoryOutput.getPosition(), 0);
+    EXPECT_EQ(threeMemoryOutput.getDataSize(), 0);
+
+    MemoryOutputStream frouMemoryOutput(memoryBlock, true);
+    EXPECT_EQ(frouMemoryOutput.getPosition(), memoryBlock.getSize());
+    EXPECT_EQ(frouMemoryOutput.getDataSize(), memoryBlock.getSize());
+
+    char *buf = (char *) malloc(sizeof(char) * 8);
+    strcpy(buf, "RocketMQ");
```

Comment:
the length of “RocketMQ” is 8,  so buf need malloc 9 bytes at least, also is exists in the following functions